server/aesdsocket.c

```diff
@@ -66,10 +95,159 @@ void *getAddress(struct sockaddr *sadd){
   return &(((struct sockaddr_in*)sadd)->sin_addr);                                                 // For IPv4 //
 }
 
+
 /*
- * Function which handles all socket communication
- * Input  : None
- * Return : int => Function returns 0 if socket communication is successful
+ * Function that would be ran as a separate thread to log time stamp values
+ * Parameters  : None  => This thread function has no input parameters
+ * Return Type : void* => Thread functoin returns NULL pointer
+ *
+ */
+void *logTimeStamp() {
+  time_t currentTime;
+  struct tm *timeInfo;
+  int returnStatus;
+  char timestamp[TIMESTAMP_LEN];
+  sleep(SLEEP_TIME_SEC);
+  
+  while (!(signalCaught)) {                                                                        // Log time stamp values till SIGTERM or SIGINT is caught //
+    time(&currentTime);
+    timeInfo = localtime(&currentTime);                                                            // Get local time //
+    strftime(timestamp, sizeof(timestamp), "timestamp:%a, %d %b %Y %T %z\n", timeInfo);            // Required time stamp format //
+    returnStatus = pthread_mutex_lock(&mutexLock);                                                 // Mutex lock before writing into file //
+    if(returnStatus != STATUS_SUCCESS){
+      syslog(LOG_ERR, "Mutex lock failed \r\n");
+    }
+    lseek(fileFd_W_R, bytesToBeWritten, SEEK_SET);                                                 // Set the cursor position //
+    bytesToBeWritten += write(fileFd_W_R, timestamp, strlen(timestamp));                           // Update number of bytes written //
+    returnStatus = pthread_mutex_unlock(&mutexLock);                                               // Mutex unlock after write //
+    if(returnStatus != STATUS_SUCCESS){
+      syslog(LOG_ERR, "Mutex Unlock failed \r\n");
+    }
+    sleep(SLEEP_TIME_SEC);                                                                         // Sleep for 10 seconds //
+  }
+  return NULL;
+}
+
+
+/*
+ * Function to exit from a thread
+ * Parameters  : 1. thread_param_t* => Struct pointer which has all the thread info 
+ *               2. char* => Pointer to the overall buffer to free(if needed)
+ * Return Type : void => Function returns nothing
+ *
+ */
+void exitThread(thread_param_t* threadParam, char* recPacket){
+  if(recPacket != NULL){
+    free(recPacket);
+    recPacket = NULL;
+  }
+  syslog(LOG_INFO, "Closed connection from %s\n", threadParam->ipString);
+  threadParam->threadComplete = true;
+  close(threadParam->clientFd);
+}
+
+
+/*
+ * Function which would be executed for different threads
+ * Parameters  : void* =>
+ * Return Type : void* => Function returns NULL pointer
+ *
+ */
+void *threadFunc(void* threadArg){
```

Comment:
@Vignesh-Vadivel I think you could simplify this function significantly if you used a pattern like this:

```
pthread_mutex_lock()
status = call_function_which_requires_mutex_lock_and_return_status()
pthread_mutex_unlock()
if (status == failed)
{
   Do error handling
}
```